mir_eval/transcription.py

```diff
@@ -0,0 +1,243 @@
+Estimation_%26_Tracking_Results_-_MIREX_Dataset#Task_2:Note_Tracking_.28NT.29):
+
+"This subtask is evaluated in two different ways. In the first setup , a
+returned note is assumed correct if its onset is within +-50ms of a ref note
+and its F0 is within +- quarter tone of the corresponding reference note,
+ignoring the returned offset values. In the second setup, on top of the above
+requirements, a correct returned note is required to have an offset value
+within 20% of the ref notes duration around the ref note's offset, or within
+50ms whichever is larger."
+
+In short, we compute precision, recall and f-measure, once without taking
+offsets into account, and the second time with.
+
+For further details see Salamon, 2013 (page 186), and references therein:
+
+    Salamon, J. (2013). Melody Extraction from Polyphonic Music Signals.
+    Ph.D. thesis, Universitat Pompeu Fabra, Barcelona, Spain, 2013.
+
+
+Conventions
+-----------
+
+Notes should be provided in the form of an interval array and a pitch list.
+The interval array contains two columns, one for note onsets and the second
+for note offsets (each row represents a single note). The pitch list contains
+the corresponding note pitch values (one value per note), represented by their
+fundamental frequency (f0) in Hertz.
+
+Metrics
+-------
+
+* :func:`mir_eval.transcription.precision_recall_f1`: The precision,
+  recall, and F-measure of the note transcription, where an estimated note is
+  considered correct if its pitch, onset and (optionally) offset are
+  sufficiently close to a reference note
+
+'''
+
+import numpy as np
+import collections
+from . import util
+import warnings
+
+
+def validate(ref_intervals, ref_pitches, est_intervals, est_pitches):
+    """Checks that the input annotations to a metric look like time intervals
+    and a pitch list, and throws helpful errors if not.
+
+    Parameters
+    ----------
+    ref_intervals : np.ndarray, shape=(n,2)
+        Array of reference notes time intervals (onset and offset times)
+    ref_pitches: list, len=n
+        List of reference pitch values in Hertz
+    est_intervals : np.ndarray, shape=(m,2)
+        Array of estimated notes time intervals (onset and offset times)
+    est_pitches : list, len=m
+        List of estimated pitch values in Hertz
+    """
+    # If reference or estimated notes are empty, warn
+    if ref_intervals.size == 0:
+        warnings.warn("Reference note intervals are empty.")
+    if len(ref_pitches) == 0:
+        warnings.warn("Reference note pitches are empty.")
+    if est_intervals.size == 0:
+        warnings.warn("Estimate note intervals are empty.")
+    if len(est_pitches) == 0:
+        warnings.warn("Estimate note pitches are empty.")
+
+    # Make sure intervals and pitches match in length
+    if not len(ref_intervals) == len(ref_pitches):
+        warnings.warn("Reference intervals and pitches have different "
```

Comment:
These should be `ValueError`s, not warnings, right? Wouldn't this mean the annotation was certainly invalid?